login-workflow/docs/authentication-workflow.md

```diff
@@ -0,0 +1,149 @@
+
+1. Create a `/src` folder in your application if it does not already exist
+2. Add an `/actions` folder inside the `src` directory.
+3. Create a file in the new `actions` directory: `AuthUIActions.tsx`
+    - The file you created, `AuthUIActions.tsx`, will handle the implementation of the authentication related actions (such as login and forgot password).
+    - You can copy this file directly from the [example](https://github.com/etn-ccis/blui-react-workflows/tree/master/login-workflow/example) project as a starting point and then update the implementation details if you choose.
+4. You might also want to copy over the `example/src/store` and `example/src/constants` folders from react-auth-workflow for the purposes of compiling with the mock `AuthUIActions` before you write your own implementation. These sample implementations make use of the browser LocalStorage, but you may want to use a different approach in order to follow best practices for cybersecurity.
+
+## Example Usage
+
+Here is an example of how you would set this up using our recommended routing solution, [React Router](https://reactrouter.com/):
+
+```tsx
+import React from 'react';
+import {
+    AuthContextProvider,
+    ContactSupportScreen,
+    ForgotPasswordScreen,
+    ResetPasswordScreen,
+} from '@brightlayer-ui/react-auth-workflow';
+import { useApp } from '../contexts/AppContextProvider';
+import { useNavigate } from 'react-router';
+import { ProjectAuthUIActions } from '../actions/AuthUIActions';
+import { Outlet, Route, Routes } from 'react-router-dom';
+import { Login } from './Login';
+import { routes } from '../navigation/Routing';
+import { useSecurityActions } from '../path-to-your-use-security-actions-implementation';
+
+export const AppRouter: React.FC = () => {
+   // Language will be managed by some state within your app, in this example, a useApp hook that gets the app language.
+    const { language, yourAppSecurityHelper } = useApp();
+    const navigate = useNavigate();
+    const securityContextActions = useSecurityActions();
+    return (
+        <Routes>
+            {/* AUTH ROUTES */}
+            <Route
+                element={
+                    <AuthContextProvider
+                        actions={ProjectAuthUIActions(yourAppSecurityHelper)}
+                        language={language}
+                        navigate={navigate}
+                        routeConfig={routes}
+                        i18n={yourI18nAppInstance}
+                        rememberMeDetails={{ email: rememberMe ? email : '', rememberMe: rememberMe }}
+                    >
+                        <ReactRouterGuestGuard isAuthenticated={app.isAuthenticated} fallBackUrl={'/'}>
```

Comment:
Recently, we have decided not to wrap <Outlet />, using ReactRouterGuestGuard for auth routes, because it will not allow routes which needs ReactRouterAuthGuard. We will be removing this ReactRouterGuestGuard wrapper and individually wrap each Route using ReactRouterGuestGuard or ReactRouterAuthGuard. This is still in progress.

So, this will look like:

```tsx
              <Route
                element={
                    <AuthContextProvider
                        actions={ProjectAuthUIActions(app)}
                        language={language}
                        navigate={navigate}
                        routeConfig={routes}
                        i18n={i18nAppInstance}
                        rememberMeDetails={{ email: rememberMe ? email : '', rememberMe: rememberMe }}
                    >
                        <Outlet />
                    </AuthContextProvider>
                }
            >
                <Route
                    path={'/login'}
                    element={
                        <ReactRouterGuestGuard isAuthenticated={app.isAuthenticated} fallBackUrl={'/'}>
                            <Login />
                        </ReactRouterGuestGuard>
                    }
                />
               <Route
                    path={'/change-password'}
                    element={
                        <ReactRouterAuthGuard isAuthenticated={app.isAuthenticated} fallBackUrl={'/login'}>
                            <ChangePasswordDialog open />
                        </ReactRouterAuthGuard>
                    }
                />
           </Route>